modules/auxiliary/scanner/winrm/winrm_cmd.rb

```diff
@@ -33,7 +33,8 @@ def initialize
       [
         OptEnum.new('WinrmAuth', [true, 'The Authentication mechanism to use', Msf::Exploit::Remote::AuthOption::AUTO, Msf::Exploit::Remote::AuthOption::WINRM_OPTIONS]),
         OptString.new('WinrmRhostname', [false, 'The rhostname which is required for kerberos']),
-        OptAddress.new('DomainControllerRhost', [false, 'The resolvable rhost for the Domain Controller'])
+        OptAddress.new('DomainControllerRhost', [false, 'The resolvable rhost for the Domain Controller']),
+        OptPath.new('WinrmKrb5Ccname', [false, 'The ccache file to use for kerberos authentication', ENV.fetch('WINRMKRB5CCNAME', ENV.fetch('KRB5CCNAME', nil))], conditions: %w[ WinrmAuth == kerberos ])
```

Comment:
The datastore validation fails when using Kerberos authentication with a CCache file and without `PASSWORD`, which should not be necessary:
```
msf6 auxiliary(scanner/winrm/winrm_cmd) > run verbose=true RHOSTS=10.0.0.4 WinrmAuth=kerberos DOMAIN=mylab.local USERNAME=Administrator WinrmRhostname=dc01.mylab.local DomainControllerRhost=10.0.0.4 WinrmKrb5Ccname=/tmp/Administrator.ccache

[-] Msf::OptionValidateError The following options failed to validate: PASSWORD
```

Reply:
I was able to fix this instance though because this module is just a scanner and *not* a login scanner.